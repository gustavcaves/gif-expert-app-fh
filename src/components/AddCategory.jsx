import { useState } from "react"


export const AddCategory = ( { onNewCategory } ) => {

    const [InputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) => {
        // console.log(target.value)
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (InputValue.trim().length <= 1) return

        // setCategories((categories) => [InputValue, ...categories])
        onNewCategory(InputValue.trim())
        setInputValue('');
       
    }



    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={ InputValue }
                // onChange={ (event) => onInputChange(event) }
                onChange={ onInputChange }
                
            />
        </form>
  )
}
