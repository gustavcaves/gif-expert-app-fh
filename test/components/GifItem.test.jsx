import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";



describe('Pruebas sobre el <GifItem /> ', () => {

    
    const title = 'Saitama';
    const url = 'https://one-push.com/saitama.jpg'


    test('debe hacer match con el snapshot', () => {
        
        const {cointainer} = render( <GifItem title={title} url={url}/> )
        expect( cointainer ).toMatchSnapshot();

    });

});