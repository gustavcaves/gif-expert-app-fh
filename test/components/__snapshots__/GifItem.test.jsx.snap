// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el <GifItem />  debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-push.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
